autoscaler/fake_kubernetes_client.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscaler
+
+import (
+	"fmt"
+)
+
+var _ = K8sClientInterface(&FakeK8sClient{})
+
+// FakeK8sClient implements K8sClientInterface
+type FakeK8sClient struct {
```

Comment:
the typical nomenclature is "mock" not "fake", unless this is the standard in our k8s codebase.